### PR TITLE
feature/ integrating new signup flow

### LINKED ISSUE #322: Integrated new Signup Flow
## Issue
When signing in with GitHub send the page href in query params as `&state=`
- RFC - https://github.com/Real-Dev-Squad/website-backend/issues/688

### Description
When you visit the `my-site` without login in, it will redirect you to github OAuth, but after login it redirects you back to `www-site` which is not what we want.

### Expected Flow
- User visits `my-site` for the first time without login.
- Now, they are redirected to the GitHub login page
- once the login is successful the user is expected to land back to the `my-site`
- Example PR - https://github.com/Real-Dev-Squad/website-www/pull/277

### PR DESCRIPTION
**Pull request by Dev-AkshatGupta**
### Issue Resolved
Closes #322

### What is the change?

- Storing the origin URL in the state, redirecting the user to the origin after successful login.
- If the origin URL is missing redirect to home.

### *Dev Tested?
 Yes

### *Tested on:

**Platforms**
Web

**Browsers**
Chrome

### CODE REVIEW

**Review by ivinayakg — APPROVED** [review]
LGTM 👍

File: app/routes/index.js [hunk 1]
```diff
@@ -22,14 +22,13 @@ export default class IndexRoute extends Route {
...
-        setTimeout(
-          () =>
-            window.open(
-              'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97',
-              '_self'
-            ),
-          2000
-        );
+        setTimeout(() => {
+          let authUrl = `https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97`;
```
Comment (rohan09-raj) [inline app/routes/index.js:26]: Can we please store this URL in some constant and then use it here?

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: app/routes/index.js (hunk 1 above)
Comment (ivinayakg) [reply app/routes/index.js:26]: Akshat check my pr for reference, try to use the same variable name that there is in the pr.

Comment (Dev-AkshatGupta) [reply app/routes/index.js:26]: Sure will look at that. Thanks for reviewing.

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**